client/components/status/status.test.js

```diff
@@ -1674,103 +1674,116 @@ describe("<Status /> interactions", () => {
   it("test getUserRadiusUsage method", async () => {
     jest.spyOn(toast, "error");
     jest.spyOn(toast, "dismiss");
-    axios
-      .mockImplementationOnce(() =>
-        Promise.reject({
-          response: {
-            status: 500,
-            headers: {},
-          },
-        }),
-      )
-      .mockImplementationOnce(() =>
-        Promise.resolve({
-          status: 200,
-          statusText: "OK",
-          data: {
-            checks: [
-              {
-                attribute: "Max-Daily-Session",
-                op: ":=",
-                value: "10800",
-                result: 0,
-                type: "seconds",
-              },
-            ],
-          },
-          headers: {},
-        }),
-      )
-      .mockImplementationOnce(() =>
-        Promise.resolve({
-          status: 200,
-          statusText: "OK",
-          data: {
-            plan: {
-              id: "d5bc4d5a-0a8c-4e94-8d52-4c54836bd013",
-              name: "Free",
-              currency: "EUR",
-              is_free: true,
-              expire: null,
-              active: true,
-            },
-          },
-          headers: {},
-        }),
-      )
-      .mockImplementationOnce(() =>
-        Promise.resolve({
-          status: 200,
-          statusText: "OK",
-          data: {
-            checks: [
-              {
-                attribute: "Max-Daily-Session",
-                op: ":=",
-                value: "10800",
-                result: 10800,
-                type: "seconds",
-              },
-            ],
-            plan: {
-              id: "d5bc4d5a-0a8c-4e94-8d52-4c54836bd013",
-              name: "Free",
-              currency: "EUR",
-              is_free: true,
-              expire: null,
-              active: true,
-            },
-          },
-          headers: {},
-        }),
-      )
-      .mockImplementationOnce(() =>
-        Promise.reject({
-          response: {
-            status: 401,
-            headers: {},
-          },
-        }),
-      );
+    jest.spyOn(global, "setTimeout");
     props = createTestProps();
     wrapper = shallow(<Status {...props} />, {
       context: {setLoading: jest.fn()},
       disableLifecycleMethods: true,
     });
     jest.spyOn(wrapper.instance(), "getUserRadiusUsage");
+
+    // Retry when server responds with 500 error
+    axios.mockImplementationOnce(() =>
+      Promise.reject({
+        response: {
+          status: 500,
+          headers: {},
+        },
+      }),
+    );
     wrapper.instance().getUserRadiusUsage();
     await tick();
     expect(wrapper.instance().state.radiusUsageSpinner).toBe(true);
+    expect(setTimeout).toHaveBeenCalledWith(expect.any(Function), 10000);
```

Comment:
What does this test line mean? A comment reminding us would be useful. I don't understand this.

Reply:
In https://github.com/openwisp/openwisp-wifi-login-pages/pull/858/commits/7a528000f9e1033a9a56b89255917331ef4043cf, I have improved the code and test for better readability.